IPython/core/tests/test_ultratb.py

```diff
@@ -38,16 +37,12 @@ def recursionlimit(frames):
 
     def inner(test_function):
         def wrapper(*args, **kwargs):
-            _orig_rec_limit = ultratb._FRAME_RECURSION_LIMIT
-            ultratb._FRAME_RECURSION_LIMIT = 50
-
             rl = sys.getrecursionlimit()
             sys.setrecursionlimit(frames)
             try:
                 return test_function(*args, **kwargs)
             finally:
                 sys.setrecursionlimit(rl)
-                ultratb._FRAME_RECURSION_LIMIT = _orig_rec_limit
```

Comment:
I believe those were here to speedup the test suite by a lot (and/or failure on CI). Is that handled by stack_data ?

I'm happy if it needs to be removed for other reasons.

Reply:
The speedup is provided by `sys.setrecursionlimit`. `ultratb._FRAME_RECURSION_LIMIT` was used for detecting recursion, although I don't know why it didn't use `sys.getrecursionlimit`.

stack_data doesn't detect recursion specifically, it detects any string of repeating frames, similar to the Python traceback.